torch/distributed/elastic/agent/server/api.py

```diff
@@ -546,7 +486,16 @@ def _rendezvous(self, worker_group: WorkerGroup) -> None:
         spec = worker_group.spec
 
         with self.record_duration("RENDEZVOUS"):
-            store, group_rank, group_world_size = spec.rdzv_handler.next_rendezvous()
+            rdzv_info = spec.rdzv_handler.next_rendezvous()
+        store = rdzv_info.store
+        group_rank = rdzv_info.rank
+        group_world_size = rdzv_info.world_size
+
+        # master_addr/master_port could be explicitly overriden
+        # TODO: BC - specific to static rdzv and can be simplifed further
+        master_addr = spec.master_addr or rdzv_info.bootstrap_store_info.master_addr
```

Comment:
should we be checking use_agent_store ?

Reply:
No, at this time it just needs master_addr and master_port regardless if store is shared or not